### PR TITLE
Fix #8: Tests suite

### LINKED ISSUE #8: Test suite
## I expect
- test of main.py
- test of Checkov
- test of Safety

### PR DESCRIPTION
**Pull request by DrPlumcake**
## I expect:
- Implemented test to Work

## Current:
- test_bandit.py/test_errors fails

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,12 +10,13 @@ ADD entrypoint.sh /
...
 
 RUN mkdir parse_scripts
-ADD parse_scripts/* /parse_scripts
+COPY parse_scripts/* /parse_scripts
 ADD request.py /
+COPY tests /
 
 RUN chmod +x /entrypoint.sh
 RUN chmod +x /sast_to_log.py
 RUN chmod +x /parse_scripts/*
```
Comment (ioggstream) [inline Dockerfile]: We don't need it
```suggestion
```

File: Dockerfile [hunk 2]
```diff
@@ -10,12 +10,13 @@ ADD entrypoint.sh /
 ADD sast_to_log.py /
 
 RUN mkdir parse_scripts
-ADD parse_scripts/* /parse_scripts
+COPY parse_scripts/* /parse_scripts
 ADD request.py /
+COPY tests /
```
Comment (ioggstream) [inline Dockerfile]: We don't need tests in our container.

File: main.py [hunk 3]
```diff
@@ -24,21 +24,61 @@
 log = logging.getLogger(__name__)
 
 
-def env_json(tool, environ=environ):
-    value_tool = json_arg_dict.get(tool, "None")
+def env_json(dict, tool, environ=environ):
```
Comment (ioggstream) [inline main.py]: Don't override the `dict` class :) 
A variable name should describe its content, not its type.

File: main.py [hunk 4]
```diff
@@ -24,21 +24,61 @@
...
+    value_tool = dict.get(tool, "None")
```
Comment (ioggstream) [inline main.py]: Why are you checking for the string "None" ?

File: main.py [hunk 5]
```diff
@@ -24,21 +24,61 @@
...
+            res = gh(
+                u_post,
+                method="POST",
+                data=tool_checks,
+                token=environ["GITHUB_TOKEN"],
+            )
+            log.info("Request Status: %s %s %s", res.status_code, res.content, res.url)
+
+        if not local:
+            log.setLevel(logging.DEBUG)
```
Comment (ioggstream) [inline main.py]: An application should not modify its logging level. The caller must decide the logging level.

File: tests/test_bandit.py [hunk 6]
```diff
@@ -19,18 +22,13 @@ def test_errors():
 
 
 def test_annotations():
-    results = json.loads(Path("tests/json/bandit.json").read_text())
-    annotations = bandit_annotations(results)
+    results = json.loads(Path(JSON_DIR / "bandit.json").read_text())
+    annotations = bandit.bandit_annotations(results)
     assert annotations[0]["path"] == "canary.py"
```
Comment (ioggstream) [inline tests/test_bandit.py:27]: OK!

File: tests/test_checkov.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+from pathlib import Path
+
+from parse_scripts import checkov
+
+TEST_DIR = Path(__file__).parent
+JSON_DIR = TEST_DIR / "json"
+
+
+def test_checkov_parse():
+    expected_comments = {
```
Comment (ioggstream) [inline tests/test_checkov.py]: OK, but this can even go in a separate test file.

**Review by ioggstream — CHANGES_REQUESTED** [review]
@DrPlumcake Generally ok provided the following suggestions.

:warning: don't override `dict` and other built-in classes!

**Comment by DrPlumcake** [comment]
@ioggstream "test_errors" of test_bandit.py still goes into failure. Suggestions?

**Comment by DrPlumcake** [comment]
`Syntax Error` does not have anymore `"invalid syntax"` string in the `args` parameter (i checked).
So, if code catches a syntax error, we directly write in the title of the error the string "invalid syntax"

**Pull request merged**